server/src/main/java/org/jetbrains/bsp/bazel/server/BloopExporter.java

```diff
@@ -0,0 +1,549 @@
+    return opt.fold(scala.Option::empty, Some::apply);
+  }
+
+  private Seq<Path> fixupClassPath(Seq<URI> input) {
+    return input
+        .iterator()
+        .map(p -> this.localArtifacts.getOrElse(p, p))
+        .distinct()
+        .map(Paths::get)
+        .toArray();
+  }
+
+  private Option<Config.Platform> extractPlatformFromLanguageData(LanguageData languageData) {
+    var javaModule = javaModuleFromLanguageData(languageData);
+    return javaModule.map(
+        mod -> {
+          var runtimeJdk = mod.runtimeJdk().getOrElse(mod.jdk());
+          var jvmConfig =
+              new Config.JvmConfig(
+                  toScalaOption(runtimeJdk.javaHome().map(Paths::get)),
+                  CollectionConverters.asScala(mod.jvmOps()).toList());
+          var classPath =
+              CollectionConverters.asScala(fixupClassPath(mod.runtimeClasspath())).toList();
+
+          return new bloop.config.Config$Platform$Jvm(
+              jvmConfig,
+              toScalaOption(mod.mainClass()),
+              scala.Some.apply(jvmConfig),
+              scala.Some.apply(classPath),
+              scala.Option.empty());
+        });
+  }
+
+  private Config.Test bloopTestFramework() {
+    var framework =
+        new Config.TestFramework(
+            CollectionConverters.asScala(
+                    Collections.singleton("munit.internal.junitinterface.PantsFramework"))
+                .toList());
+    return new Config.Test(
+        CollectionConverters.asScala(Collections.singleton(framework)).toList(),
+        new Config.TestOptions(List$.MODULE$.empty(), List$.MODULE$.empty()));
+  }
+
+  private scala.collection.immutable.List<String> convertTags(io.vavr.collection.Set<Tag> tags) {
+    Set<String> ret;
+    if (tags.contains(Tag.TEST)) {
+      ret = HashSet.of("test");
+    } else {
+      ret = HashSet.of("library");
+    }
+
+    return CollectionConverters.asScala(ret).toList();
+  }
+
+  private Config.Resolution extractResolution(Module module) {
+    var resolutionModules =
+        module
+            .sourceDependencies()
+            .map(
+                sourceDep -> {
+                  var artifact =
+                      new Config.Artifact(
+                          "",
+                          scala.Option.apply("sources"),
+                          scala.Option.empty(),
+                          Paths.get(sourceDep));
+                  return new Config.Module(
+                      "",
+                      "",
+                      "",
+                      scala.Option.empty(),
+                      CollectionConverters.asScala(Collections.singleton(artifact)).toList());
+                });
+    return new Config.Resolution(CollectionConverters.asScala(resolutionModules).toList());
+  }
+
+  private <T> scala.collection.immutable.List<T> emptyList() {
+    return List$.MODULE$.empty();
+  }
+
+  private Option<scala.collection.immutable.List<Config.SourcesGlobs>> reGlob(
```

Comment:
plus it's a huge method which reallllllyyy needs tests (and which needs to be smaller and in a separate class)

Reply:
+100 on tests!  Once things stabilize I'll definitely write a bunch of tests for it.